### ISSUE TITLE
Function that takes callable with generics

### ISSUE THREAD

**Issue by erictraut** [microsoft/pyright#4920]
### Discussed in https://github.com/microsoft/pyright/discussions/4919

<div type='discussions-op-text'>

<sup>Originally posted by **sylee957** April  6, 2023</sup>
I can't match argument `tuple[X] -> Y` with `tuple[int] -> int` function in the following example.

```python3
from collections.abc import Callable
from typing import TypeVar

X = TypeVar("X")
Y = TypeVar("Y")


def unwrap(func: Callable[[tuple[X]], Y]) -> Callable[[X], Y]:
    def wrapped(x: X) -> Y:
        return func((x,))

    return wrapped


def f(x: tuple[int]) -> int:
    return x[0] + 1
```

`unwrap(f)` does not pass the type check.</div>

**Comment by erictraut** [comment]
This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.303, which I just published. The change will be included in a future version of pylance.
